src/client.ts

```diff
@@ -309,8 +306,18 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
       }, timeout) as any;
     });
 
-    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(res => {
+    return Promise.race([this.dataReadyPromise, timeoutPromise]).then(async res => {
       clearTimeout(timeoutId);
+      if (res.success) {
+        const isSegmentsFetched = await this.fetchQualifiedSegments();
```

Comment:
Can you confirm if JS SDK returns true when ODP is not integrated or disabled. It should continue to work as success for backward compatibility.

Reply:
I see `fetchQualifiedSegments` called both from setUser and here. Wondering if they are not called twice when SDK starts. Can we add a comment about the flow?

Reply:
it does return true when ODP is not integrated or disabled.

Reply:
for the flow, to initialize the sdk we need to call the js onready which is awaited on the onready method. so for the initial call we need to call segment fetch on onready and the on the setuser segment call will execute on any following setuser calls after initialization. 
and there is a check on setuser line:387 so that during initializing segment fetch is not called twice.